src/types/MarkdownTextRenderer.ts

```diff
@@ -38,7 +38,9 @@ export class MarkdownTextRenderer {
 				} else {
 					markdownText += this.buildMarkdownLinkString(file.name, null, indentLevel, true)
 				}
-				markdownText += this.buildStructureMarkdownText(this.buildFileTree(children), indentLevel + 1)
+				if (indentLevel < this.plugin.settings.headlineDepth) {
```

Comment:
Filters out any things indented more than `headerlineDepth` which does not nessesarily equal the header level